integration/auto-wake_test.go

```diff
@@ -0,0 +1,209 @@
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/okteto/okteto/pkg/okteto"
+)
+
+func TestAutoWake(t *testing.T) {
+	tName := fmt.Sprintf("Test-%s", runtime.GOOS)
+	ctx := context.Background()
+	oktetoPath, err := getOktetoPath(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := exec.LookPath(kubectlBinary); err != nil {
+		t.Fatalf("kubectl is not in the path: %s", err)
+	}
+
+	name := strings.ToLower(fmt.Sprintf("%s-%d", tName, time.Now().Unix()))
+	namespace := fmt.Sprintf("%s-%s", name, user)
+
+	dir, err := os.MkdirTemp("", tName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	log.Printf("created tempdir: %s", dir)
+
+	dPath := filepath.Join(dir, "deployment.yaml")
+	if err := writeDeployment(deploymentTemplate, name, dPath); err != nil {
+		t.Fatal(err)
+	}
+
+	sfsPath := filepath.Join(dir, "sfs.yaml")
+	if err := writeStatefulset(name, sfsPath); err != nil {
+		t.Fatal(err)
+	}
+
+	contentPath := filepath.Join(dir, "index.html")
+	if err := os.WriteFile(contentPath, []byte(name), 0644); err != nil {
+		t.Fatal(err)
+	}
+
+	log.Printf("original content: %s", name)
+
+	manifestPath := filepath.Join(dir, "okteto.yml")
+	if err := writeManifest(manifestPath, name); err != nil {
+		t.Fatal(err)
+	}
+
+	stignorePath := filepath.Join(dir, ".stignore")
+	if err := os.WriteFile(stignorePath, []byte("venv"), 0600); err != nil {
+		t.Fatal(err)
+	}
+
+	startNamespace := getCurrentNamespace()
+	defer changeToNamespace(ctx, oktetoPath, startNamespace)
+	if err := createNamespace(ctx, oktetoPath, namespace); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := deploy(ctx, namespace, name, dPath, true); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := deploy(ctx, namespace, name, sfsPath, true); err != nil {
+		t.Fatal(err)
+	}
+
+	client, err := okteto.NewOktetoClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	endpoint := fmt.Sprintf("https://%s-%s.%s", name, namespace, appsSubdomain)
+	content, err := getContent(endpoint, 150, nil)
+	if err != nil {
+		t.Fatalf("failed to get content: %s", err)
+	}
+	if content == "" {
+		t.Fatalf("failed to get content")
+	}
+
+	if err := client.SleepNamespace(ctx, namespace); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := checkIfSleeping(ctx, name, namespace); err != nil {
+		t.Fatal(err)
+	}
+
+	content, err = getContent(endpoint, 150, nil)
+	if err != nil {
+		t.Fatalf("failed to get content: %s", err)
+	}
+	if content == "" {
+		t.Fatalf("failed to get content")
+	}
+
+	if err := checkIfAwake(ctx, name, namespace); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := client.SleepNamespace(ctx, namespace); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := checkIfSleeping(ctx, name, namespace); err != nil {
+		t.Fatal(err)
+	}
+
+	var wg sync.WaitGroup
+	upErrorChannel := make(chan error, 1)
+	p, err := up(ctx, &wg, namespace, name, manifestPath, oktetoPath, upErrorChannel)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	content, err = getContent(endpoint, 150, nil)
```

Comment:
I wouldn't call either `getContent` or `down`. Just up should wake up everything

Reply:
+1 to not call getContent, but if you don't call down there will be a zombie process running on background that is using local port 8080

Reply:
we can call it after checking the namespace is active